src/index.ts

```diff
@@ -1995,7 +1995,7 @@ export class GossipSub extends EventEmitter<GossipsubEvents> implements PubSub<G
    *
    * For messages not from us, this class uses `forwardMessage`.
    */
-  async publish(topic: TopicStr, data: Uint8Array): Promise<PublishResult> {
+  async publish(topic: TopicStr, data: Uint8Array, allowPublishToZeroPeers = false): Promise<PublishResult> {
```

Comment:
```suggestion
  type PublishOpts {
    allowPublishToZeroPeers?: boolean
  }

  async publish(topic: TopicStr, data: Uint8Array, opts?: PublishOpts): Promise<PublishResult> {
    const allowPublishToZeroPeers = opts?.allowPublishToZeroPeers ?? this.opts.allowPublishToZeroPeers;
    
    ...
    
    if (
      tosend.size === 0 &&
      !allowPublishToZeroPeers
      !willSendToSelf
    ) {
      throw Error('PublishError.InsufficientPeers')
  }
```

Reply:
I added the `PublishOpts` type to the [types](https://github.com/ChainSafe/js-libp2p-gossipsub/pull/395/files#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR75) file.